index.js

```diff
@@ -3,6 +3,50 @@
 const fp = require('fastify-plugin')
 var pg = require('pg')
 
+function transactionUtil (pool, fn, cb) {
+  pool.connect((err, client, done) => {
+    if (err) return cb(err)
+
+    const shouldAbort = (err) => {
+      if (err) {
+        client.query('ROLLBACK', () => {
+          done()
+        })
+      }
+      return !!err
+    }
+
+    client.query('BEGIN', (err) => {
+      if (shouldAbort(err)) return cb(err)
+
+      fn(client).then(res => {
```

Comment:
Can you pass a callback as well and support the dual promise/callback pattern?

Reply:
I am sorry I thought over this but am still confused. What will that callback look like? How do I know the user wants a callback Could you please give me some pointers?

Reply:
Here is an example (from avvio) https://github.com/mcollina/avvio/blob/9f947aa5a71db44f185dfab16fdf75709d00dcb3/plugin.js#L82